### DISCUSSION TITLE
latency calculation

### DISCUSSION THREAD

**Discussion by Bhargavnaidu-zelar** [grafana/tempo#4116] [General]
Hey team, how can we know the aggregation for traces to observe the latency, whether it is correct or not?

**Comment by joe-elliott** [comment]
I'm not quite sure what aggregation you're talking about. Regardless, if you would like to verify the correctness I suppose you can review the code or validate it externally.

Here's an example of a CLI tool that uses our internal schema:
https://github.com/stoewer/parquet-cli

You could use this to review the contents of the blocks at a low level which might help your validation.

**Reply by Bhargavnaidu-zelar** [reply]
i have been talking about how does the latency being calculated for each span in the background while application is running?

**Reply by mapno** [reply]
That depends on the instrumentation and environment is it run on, but generally it'll use the machines clock to time start and end.

Pseudocode
```
{
  // start span - time 50
  t0 := time.Now()

  // do work ...
  work()

  // finish span - time 60
  t1 := time.Now()
  // duration = t1 - t0 = 10
}
```
